### ISSUE TITLE
 the seq_kmer value  set to 5 for R9.4.1 data

### ISSUE THREAD

**Issue by persuelmk** [ZKI-PH-ImageAnalysis/seq2squiggle#24]
Hi, I hope you’re doing well!

In the CONFIG.md file, the seq_kmer value is set to 5 for R9.4.1 data. However, according to ONT's official documentation, the R9.4.1 pore model uses 6-mers. Could you please clarify why the seq_kmer is set to 5 for R9.4.1 data in the configuration? Is there a specific reason for this choice, or are there any considerations I might be overlooking?

Thanks a lot for your time and help!

**Comment by denisbeslic** [comment]
Hi!
 
Thank you for catching that! That was indeed just a typo on my part, which I’ve now corrected.  
As you correctly noted, the pore model (and the uncalled4 TSV output used to train seq2squiggle) is based on 6-mers. This is also reflected in the seq2squiggle codebase, where seq_kmer is set to 6. See [here](https://github.com/ZKI-PH-ImageAnalysis/seq2squiggle/blob/a69d320988134298c9ff131d3ac46fc2b900a689/src/seq2squiggle/utils.py#L245-L263)

**Comment by persuelmk** [comment]
<img width="1849" height="50" alt="Image" src="https://github.com/user-attachments/assets/be46ac26-9b38-4a03-8731-6f61f50674d5" />    

![Image](https://github.com/user-attachments/assets/cc92d419-290a-4f86-9344-1987c96b11aa)
I hope this message finds you well. I recently trained a custom model using the configuration file set to 6-mer, as per the standard in the seq2squiggle codebase. However, when I used my trained model (last1.ckpt) to simulate electrical signals, the resulting signal values appeared to be in an abnormal range.

Could you kindly advise on what might be causing this issue? I suspect it could be related to the training process, the model's parameters, or perhaps some overlooked settings in the configuration. Any insights you could provide would be greatly appreciated.

Thank you very much for your time and for developing such a valuable tool!

**Comment by denisbeslic** [comment]
Thanks for your interest in the tool!

That's hard to say without seeing the training data. On how much data did you train? Did the training loss decrease over time? 
If training looked normal, my first guess would be it's related to data scaling. What input data did you use for training (e.g., Uncalled 4, f5c)? Can you check the scale of your input.tsv and .npy files?

```
head -10 input.tsv
```

```python
import numpy as np
data = np.load('chunks.npy', mmap_mode='r')
print(data[:10])
```

**Comment by persuelmk** [comment]
hi!I'm having two issues while using your tools and would appreciate your advice:

Standardization Problem:
Input: events.tsv from Uncalled4 alignment
Command: python3 standardize-events.py events.tsv events_norm.tsv --sigtk sigtk.txt
Issue: The signal ranges in input and output files are identical

Training Plots Missing:
Used --save_valid_plot in seq2squiggle train but can't find loss function plots

**Comment by denisbeslic** [comment]
Standardization:
Does your sigtk.txt look as required in the README? Does the standardize-events.py terminate successfully? Can you print
```
head -10 events.tsv 
head -10 events_norm.tsv 
head -10 sigtk.txt
```

Training Plots Missing:
Loss function will not be plotted, only a few example plots. For tracking the loss function I used wandb, which you can also activate on your own account by setting wandb_logger_state in the config to True and [logging ](https://docs.wandb.ai/models/quickstart) with your wandb account

**Comment by persuelmk** [comment]
<img width="1022" height="234" alt="Image" src="https://github.com/user-attachments/assets/a78aef77-3a81-406e-9177-e21846323e73" />

<img width="1024" height="231" alt="Image" src="https://github.com/user-attachments/assets/e5c75521-94d5-42c9-8a2c-cee86f4f25a3" />

<img width="620" height="169" alt="Image" src="https://github.com/user-attachments/assets/c2c3b76b-ab85-4d8b-87a7-581d0da30d8c" />
yes,the standardize-events.py terminate successfully.

**Comment by denisbeslic** [comment]
Can you show the samples column? The standardization only changes the samples column.
Can you send me a small subset of your input events.tsv and sigtk.txt file?

**Comment by persuelmk** [comment]
[events_norm_1.tsv](https://github.com/user-attachments/files/23578631/events_norm_1.tsv)

[events_1.tsv](https://github.com/user-attachments/files/23578633/events_1.tsv)

[sigtk.txt](https://github.com/user-attachments/files/23578488/sigtk.txt)
Hi ，I've checked the samples column—events.tsv and events_norm.tsv are identical .

Thank you for your help!

**Comment by denisbeslic** [comment]
Thanks, I removed the tabs and it worked. I updated the standardize-events.py on the dev branch to accept --mean and --std arguments to avoid such errors in the future.

However, I realized your input TSV does not need any denormalization since the samples values are already scaled in pA. This script is only needed when your samples column values are "normalized pore model values, which have a mean of 0 and a standard deviation of 1".  I guess for uncalled4, this is only the case for RNA004/R10 models. Since you already have those, you can just proceed step 3 and preprocess. This likely also caused scaling issue mentioned before.
